### PR TITLE
Example gateway applications

### PR DESCRIPTION
**Pull request by balazsracz**
Adds two gateway type applications to the example app collection:
- an LCC-to-DMX gateway that can drive stage lights using a standard industrial protocol.
- an LCC-to-MODBUS gateway that can drive industrial motor controllers (use at your own risk).

Both of these are RS485 based protocols.

### CODE REVIEW

File: include/freertos/tc_ioctl.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * This file implements can specific ioctl() keys for serial devices.
+ *
+ * @author Balazs Racz
+ * @date 28 Feb 2018
+ */
+
+#ifndef _FREERTOS_TC_IOCTL_H_
+#define _FREERTOS_TC_IOCTL_H_
+
+#define TCSBRK		0x5409
```
Comment (bakerstu) [inline include/freertos/tc_ioctl.h]: Take a look at can_ioctl.h.  It uses the IO* macros for giving the interface a magic number and encoding the transaction type.  This is useful in making sure the ioctl is not misused by the application.  This concept was borrowed from Linux.

File: src/freertos_drivers/ti/CC32xxUart.cxx [hunk 2]
```diff
@@ -119,6 +121,30 @@ void CC32xxUart::disable()
...
+ * @param data key data
+ * @return 0 upon success, -errno upon failure
+ */
+int CC32xxUart::ioctl(File *file, unsigned long int key, unsigned long data)
+{
+    switch (key)
+    {
+        default:
+            return -EINVAL;
+        case TCSBRK:
```
Comment (bakerstu) [inline src/freertos_drivers/ti/CC32xxUart.cxx]: Use the IO* macros to decode this key for correctness.

File: src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx [hunk 3]
```diff
@@ -611,6 +611,12 @@ void CC32xxWiFi::connecting_update_blinker()
 void CC32xxWiFi::set_default_state()
 {
     long result = sl_Start(0, 0, 0);
+#ifdef SL_API_V2
+    if (result == SL_ERROR_ROLE_STA_ERR) {
```
Comment (bakerstu) [inline src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx]: style: hanging brace.

File: include/freertos/tc_ioctl.h (hunk 1 above)
Comment (balazsracz) [reply include/freertos/tc_ioctl.h]: done

File: src/freertos_drivers/ti/CC32xxUart.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/ti/CC32xxUart.cxx]: I'm not sure what you mean. I'm looking at the linux kernel and they use a case for this purpose like in the code here:
https://github.com/torvalds/linux/blob/d8a332730e757129e70675679f2b2a03f1ecf65e/drivers/tty/tty_io.c#L2539

File: src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx (hunk 3 above)
Comment (balazsracz) [reply src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx]: done

File: src/freertos_drivers/ti/CC32xxUart.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/ti/CC32xxUart.cxx]: per offline conversation I'm marking this as done.

**Pull request merged**